resolve-tsconfig.js

```diff
@@ -0,0 +1,79 @@
+import { join } from 'node:path'
```

Comment:
`ts-standard` to be able to work need to define `parserOptions.project` in its `eslintrc.json`, it's mandatory to set a `tsconfig.json` file. This file is highly inspired from the current code base.

The important function to look at is `_getTSConfigPath`:
- First, it tries to find if one of the default `tsconfig.json` locations exist.
- If `--project` CLI flag is provided, we take this `tsconfig.json` path, else we take the `tsconfig.json` path provided in the `package.json` in the `ts-standard.project` property if it exists.